src/nvim/buffer.c

```diff
@@ -1719,21 +1719,11 @@ buf_T *buflist_new(char_u *ffname_arg, char_u *sfname_arg, linenr_T lnum, int fl
     buf = curbuf;
     // It's like this buffer is deleted.  Watch out for autocommands that
     // change curbuf!  If that happens, allocate a new buffer anyway.
```

Comment:
"Watch out for autocommands" is kinda out of place here now that the code isn't running autocmds.